src/commonMain/kotlin/com/eiknat/lemmyclient/api/comment/Comment.kt

```diff
@@ -13,31 +13,31 @@ object Comment {
         return HttpAPI.get(Endpoint("/comment/list"), form.toParams())
     }
 
-    suspend fun create(form: CreateCommentForm): CommentResponse {
+    suspend fun create(form: CreateCommentForm): APIResponse<CommentResponse> {
         return HttpAPI.post(Endpoint("/comment"), form)
     }
 
     suspend fun edit(form: EditCommentForm): CommentResponse {
         return HttpAPI.put(Endpoint("/comment"), form)
     }
 
-    suspend fun delete(form: DeleteCommentForm): CommentResponse {
+    suspend fun delete(form: DeleteCommentForm): APIResponse<CommentResponse> {
         return HttpAPI.post(Endpoint("/comment/delete"), form)
     }
 
-    suspend fun remove(form: RemoveCommentForm): CommentResponse {
+    suspend fun remove(form: RemoveCommentForm): APIResponse<CommentResponse> {
         return HttpAPI.post(Endpoint("/comment/remove"), form)
     }
 
-    suspend fun markAsRead(form: MarkCommentAsReadForm): CommentResponse {
+    suspend fun markAsRead(form: MarkCommentAsReadForm): APIResponse<CommentResponse> {
         return HttpAPI.post(Endpoint("/comment/mark_as_read"), form)
     }
 
-    suspend fun like(form: CommentLikeForm): CommentResponse {
+    suspend fun like(form: LikeCommentForm): APIResponse<CommentResponse> {
         return HttpAPI.post(Endpoint("/comment/like"), form)
     }
 
-    suspend fun save(form: SaveCommentForm): CommentResponse {
-        return HttpAPI.post(Endpoint("/comment/save"), form)
+    suspend fun save(form: SaveCommentForm): APIResponse<CommentResponse> {
+        return HttpAPI.put(Endpoint("/comment/save"), form)
```

Comment:
what's the different between post and put?

Reply:
the simplest explanation i have is that a PUT says "take this entity and either create it or if it exists update it" and a POST is "can you append a new entity to the collection with these details?"

so a PUT is idempotent - same call same result, but maybe a different response because the entity's state was changed.

 a POST is non-idempotent - same call, different result. each request is a new entity.

Reply:
Makes sense :)